state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2206,6 +2221,40 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			}
 		}
 
+		let primary_broker = broker_fee.account.clone();
+
+		let broker_fees =
+			if T::AccountRoleRegistry::has_account_role(&primary_broker, AccountRole::Broker) {
+				[broker_fee]
+				.into_iter()
+				.chain(affiliate_fees.into_iter().filter_map(
+					|Beneficiary { account: short_affiliate_id, bps }| {
+						if let Some(affiliate_id) =
+							T::AffiliateRegistry::lookup(&primary_broker, short_affiliate_id)
+						{
+							Some(Beneficiary { account: affiliate_id, bps })
+						} else {
+							// In case the entry not found, we ignore the entry, but process the
+							// swap (to avoid having to refund it).
+							Self::deposit_event(Event::<T, I>::UnknownAffiliateBroker {
+								broker_id: primary_broker.clone(),
+								short_affiliate_id,
+							});
+
+							None
+						}
+					},
+				))
+				.collect::<Vec<_>>()
+				.try_into()
+				.expect("must fit since affiliates are limited to 1 fewer element than beneficiaries")
+			} else {
+				Self::deposit_event(Event::<T, I>::UnknownPrimaryBroker {
+					broker_id: primary_broker,
+				});
+				Default::default()
```

Comment:
Do we have a Linear issue to deal with this ie. what to do as fallback? We should probably store the data somewhere so we can reject the deposit / trigger a swap.

Reply:
But we already do trigger a swap. We can consider refunding of course, but I figure this needs some discussion first. Storing which data are you talking about tough?

Reply:
As discussed I've opened a triage issue PRO-1804 

I think we might be able to take the same path as we do for transaction screening.